tg_bot/modules/rss.py

```diff
@@ -0,0 +1,322 @@
+import html
+
+from telegram import ParseMode, constants
+from telegram.ext import CommandHandler
+
+from tg_bot import dispatcher, SUDO_USERS, updater
+from tg_bot.modules.sql import rss_sql as sql
+from tg_bot.modules.helper_funcs.chat_status import bot_admin, user_admin
+
+from feedparser import parse
+
+
+def show_url(bot, update, args):
+    tg_chat_id = str(update.effective_chat.id)
+
+    if len(args) >= 1:
+        # there is an actual link written
+
+        tg_feed_link = args[0]
+        link_processed = parse(tg_feed_link)
+
+        if 'title' in link_processed.feed:
+            feed_title = link_processed.feed.title
+        else:
+            feed_title = "Unknown"
+
+        if 'description' in link_processed.feed:
+            feed_description = link_processed.feed.description
+        else:
+            feed_description = "Unknown"
+
+        if 'link' in link_processed.feed:
+            feed_link = link_processed.feed.link
+        else:
+            feed_link = "Unknown"
+
+        if 'title' in link_processed.entries[0]:
+            entry_title = link_processed.entries[0].title
+        else:
+            entry_title = "Unknown"
+
+        if 'description' in link_processed.entries[0]:
+            entry_description = link_processed.entries[0].description
+        else:
+            entry_description = "Unknown"
+
+        if 'link' in link_processed.entries[0].link:
```

Comment:
Logic here seems a bit strange, are you sure this is what you meant to do?

Reply:
Yup

Reply:
Can you explain? All the other if statements are different, checking `entries[0]`, not `entries[0].link`. Especially suspicious given its checking if `link` is in `link`.

Reply:
The whole code is supposed to check if there's the basic data in the RSS Feed and entry, so it won't return errors if it doesn't find the data. I forgot to remove the `.link` part in the if clause.